Corregir los types de las props en el comp. `VideoEmbedPreview`

### ¿Cuál es el bug/error que has encontrado?

El en esquema [`video`](https://github.com/Juguetear/juguetear-web/blob/c83d33c52409ff120fbd1c88d6c7c232580028f5/schemas/video.ts) de Sanity se muestra un error bajo la propiedad [`components.preview`](https://github.com/Juguetear/juguetear-web/blob/c83d33c52409ff120fbd1c88d6c7c232580028f5/schemas/video.ts#L16)

Los types de las props en el componente [`VideoEmbedPreview`](https://github.com/Juguetear/juguetear-web/blob/c83d33c52409ff120fbd1c88d6c7c232580028f5/components/VideoEmbedPreview.tsx) no son correctos.

### Si tienes una solución, ¿cuáles son los pasos necesarios para implementarla?

Crear una `interface` para el componente y extender a los types que espera la propiedad [`components.preview`](https://github.com/Juguetear/juguetear-web/blob/c83d33c52409ff120fbd1c88d6c7c232580028f5/schemas/video.ts#L16).  
La `interface` debera mantener las propiedades actuales de las props del componente (`url.url`).

> Todos los cambios se deberian realizar en el Componente y no en el esquema.

**Types que espera `preview`**
![image](https://user-images.githubusercontent.com/78808163/222783928-3a00357f-c0cc-4bd1-8ee6-8d09e78048f9.png)

### ¿Cómo se puede reproducir este error?

- Ejecutar `yarn tsc` desde la consola

### Por favor completá con logs relevantes respecto al bug.

<details>
<summary>Log completo</summary>

```log

❯ yarn tsc

schemas/video.ts:17:5 - error TS2322: Type '(value: { url: { url: string; }; }) => Element' is not assignable to type 'ComponentType<PreviewProps<PreviewLayoutKey>> | undefined'.
  Type '(value: { url: { url: string; }; }) => Element' is not assignable to type 'FunctionComponent<PreviewProps<PreviewLayoutKey>>'.
    Types of parameters 'value' and 'props' are incompatible.
      Property 'url' is missing in type 'PreviewProps<PreviewLayoutKey>' but required in type '{ url: { url: string; }; }'.

17     preview: VideoEmbedPreview,
       ~~~~~~~

  components/VideoEmbedPreview.tsx:4:37
    4 function VideoEmbedPreview(value: { url: { url: string } }) {
                                          ~~~
    'url' is declared here.


Found 1 error in schemas/video.ts:17

```

</details>

### Código de conducta y licencia.

- [x] He comprobado que no se haya reportado el issue.
- [x] Estoy de acuerdo con seguir el [código de conducta](https://github.com/Juguetear/juguetear-web/blob/development/CODE_OF_CONDUCT.md) de este proyecto.
- [x] Estoy de acuerdo con seguir las [licencias](https://github.com/Juguetear/juguetear-web/blob/development/CONTRIBUTING.md#licencias) de este proyecto.

--- a/components/VideoEmbedPreview.tsx
+++ b/components/VideoEmbedPreview.tsx
@@ -1,20 +1,15 @@
 import getYouTubeID from "get-youtube-id";
 import { PreviewLayoutKey, PreviewProps } from "sanity";
 
-type VideoSelect = {
-  url?: {
-    url: string;
-  };
-};
-
 interface VideoEmbedPreviewProps extends PreviewProps<PreviewLayoutKey> {
-  select?: VideoSelect;
+  video?: {
+    url: string | undefined;
+  };
 }
 
-const VideoEmbedPreview = (props: VideoEmbedPreviewProps) => {
-  const { select } = props;
-  const videoId =
-    !select || !select.url ? undefined : getYouTubeID(select.url?.url);
+const VideoEmbedPreview = ({ video }: VideoEmbedPreviewProps) => {
+  const url = video?.url;
+  const videoId = url ? getYouTubeID(url) : undefined;
   const YTUrl = `https://www.youtube.com/embed/${videoId}`;
 
   if (!videoId) {

--- a/components/VideoEmbedPreview.tsx
+++ b/components/VideoEmbedPreview.tsx
@@ -1,7 +1,20 @@
 import getYouTubeID from "get-youtube-id";
+import { PreviewLayoutKey, PreviewProps } from "sanity";
 
-function VideoEmbedPreview(value: { url: { url: string } }) {
-  const videoId = getYouTubeID(value.url?.url);
+type VideoSelect = {
+  url?: {
+    url: string;
+  };
+};
+
+interface VideoEmbedPreviewProps extends PreviewProps<PreviewLayoutKey> {
+  select?: VideoSelect;
+}
+
+const VideoEmbedPreview = (props: VideoEmbedPreviewProps) => {
+  const { select } = props;
+  const videoId =
+    !select || !select.url ? undefined : getYouTubeID(select.url?.url);
   const YTUrl = `https://www.youtube.com/embed/${videoId}`;
 
   if (!videoId) {
@@ -17,6 +30,6 @@ function VideoEmbedPreview(value: { url: { url: string } }) {
       allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
     ></iframe>
   );
-}
+};
 
 export default VideoEmbedPreview;

--- a/schemas/video.ts
+++ b/schemas/video.ts
@@ -17,7 +17,7 @@ export default defineType({
   },
   preview: {
     select: {
-      url: "url",
+      video: "url",
     },
   },
 });

